scene/animation/animation_mixer.cpp

```diff
@@ -1169,32 +1171,32 @@ void AnimationMixer::_blend_process(double p_delta, bool p_update_only) {
 					if (track->root_motion && calc_root) {
 						double prev_time = time - delta;
 						if (!backward) {
-							if (Animation::is_less_approx(prev_time, 0)) {
+							if (Animation::is_less_approx(prev_time, start)) {
 								switch (a->get_loop_mode()) {
 									case Animation::LOOP_NONE: {
-										prev_time = 0;
+										prev_time = start;
 									} break;
 									case Animation::LOOP_LINEAR: {
-										prev_time = Math::fposmod(prev_time, (double)a_length);
+										prev_time = Math::fposmod(prev_time - start, end - start) + start;
```

Comment:
```suggestion
										prev_time = Math::fposmod(prev_time, end - start) + start;
```
> Still investigating, but at first look, it appears this PR might have caused regressions with root motion calculation and will need to be fixed before merging.

Nice pick, perhaps the calculation regarding prev_time is incorrect.

Reply:
Sorry for the late reply. I suppose what you mean here is that the root motion should not be calculated based on the current section, but instead always be based on the full length of the animation? (So `start == 0` and `end == (double) a_length`)